front/app/containers/Admin/messaging/index.tsx

```diff
@@ -31,6 +30,36 @@ const MessagingDashboard = () => {
   });
   const textingEnabled = useFeatureFlag({ name: 'texting' });
 
+  useEffect(() => {
+    if (pathname.match(/\/admin\/messaging$/)) {
+      return;
+    }
+
+    const redirect = (url: string) => {
+      clHistory.replace({
+        pathname: url,
+        search: window.location.search,
+      });
+    };
+
+    if (canManageManualCampaigns && manualEmailingEnabled) {
+      return redirect('/admin/messaging/emails/custom');
+    }
+    if (canManageAutomatedCampaigns && automatedEmailingEnabled) {
+      return redirect('/admin/messaging/emails/automated');
+    }
+    if (textingEnabled) {
+      return redirect('/admin/messaging/texting');
+    }
```

Comment:
Where does it come from? Does it fix some bug?

Reply:
This was just extremely bad code that I had to fix. Before, we were redirecting inside of the render function, which should never be done and can easily lead to infinite loops and page crashes.